src/index.ts

```diff
@@ -1,176 +1,102 @@
-import {
-  AccountId,
-  TokenId,
-  TopicId,
-} from '@hashgraph/sdk';
-import keccak256 from 'keccak256';
-import {
-  API_MAX_LIMIT,
-  NameHash,
-  SLDTopicMessage,
-  TLDTopicMessage,
-} from './config/constants.config';
-import {
-  queryNFTOwner,
-  querySLDTopicMessages,
-  queryTLDTopicMessages,
-} from './contract.utils';
-
-export class HashgraphNames {
-  tldMessages: TLDTopicMessage[];
-
-  constructor() {
-    this.tldMessages = [];
-    this.populateTLDMessages();
+import { MirrorNode, NetworkType } from './mirrorNode';
+import { hashDomain } from './hashDomain';
+import { TopLevelDomain } from './types/TopLevelDomain';
+import { SecondLevelDomain } from './types/SecondLevelDomain';
+import { MessageObject } from './types/MessageObject';
+import { NameHash } from './types/NameHash';
+
+const tldTopicId = '0.0.48097305';
+
+export class Resolver {
+  mirrorNode: MirrorNode;
+  topLevelDomains: TopLevelDomain[] = [];
+
+  constructor(networkType: NetworkType, authKey = '') {
+    this.mirrorNode = new MirrorNode(networkType, authKey);
   }
 
-  /**
- * @description Queries the Manager Topic for all TLD messages and stores them
- */
-  populateTLDMessages = async () => {
-    this.tldMessages = await queryTLDTopicMessages();
-  };
+  public async init() {
+    this.topLevelDomains = await this.getTopLevelDomains();
+  }
 
   /**
- * @description Generate a NameHash of the provided domain string
- * @param domain: {string} The domain string to hash
- * @returns {Buffer}
+ * @description Resolves a Second Level Domain to the wallet address of the domain's owner
+ * @param domain: {string} The domain to query
+ * @returns {Promise<AccountId>}
  */
-  static generateNameHash = (domain: string): NameHash => {
-    if (!domain) {
-      return {
-        domain,
-        tldHash: Buffer.from([0x0]),
-        sldHash: Buffer.from([0x0]),
-      };
-    }
-    const domainsList = domain.split('.').reverse();
-    const tld = domainsList[0];
-    let sld;
-    if (domainsList.length > 1) {
-      sld = domainsList.slice(0, 2);
+  public async resolveSLD(domain: string): Promise<string> {
+    try {
+      const nameHash = hashDomain(domain);
+      const sld = await this.getSecondLevelDomain(nameHash);
+      const [tokenId, serial] = sld.nftId.split(':');
+      const nft = await this.mirrorNode.getNFT(tokenId, serial);
+      return nft.account_id;
+    } catch (err) {
+      throw err;
```

Comment:
if we arent doing anything except catching and rethrowing we dont need to catch it.